### PR TITLE
Rename "stages" to "lessons" in i18n: Step 4

### PR DESCRIPTION
**Pull request by Hamms**
Overall plan of attack:

1. Add all strings keyed by the term "stage" to crowdin under the key "lesson". (original: https://github.com/code-dot-org/code-dot-org/pull/34854, second attempt: https://github.com/code-dot-org/code-dot-org/pull/35408)
2. Wait for a sync; Crowdin will automatically duplicate translations for the new strings (https://github.com/code-dot-org/code-dot-org/pull/35426)
3. Update all I18n.t references currently using the key "stage" to use the new "lesson" translations (original: https://github.com/code-dot-org/code-dot-org/pull/34878, second attempt: https://github.com/code-dot-org/code-dot-org/pull/35407)
4. Remove old "stage" strings now that they are no longer used (**this PR**)

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/dsl/script_dsl.rb [hunk 1]
```diff
@@ -270,22 +270,18 @@ def endvariants
 
   # @override
   def i18n_hash
-    i18n_stage_strings = {}
+    i18n_lesson_strings = {}
     @lessons.each do |stage|
```
Comment (uponthesun) [inline dashboard/app/dsl/script_dsl.rb]: Want to get this `stage` variable as well?

**Review by uponthesun — COMMENTED** [review]
There's a few more instances of `stage:` in `scripts.en.yml`, are those from something else? Do they matter?

```
Winters-MBP:code-dot-org winterdong$ grep -n "stage\:" dashboard/config/locales/scripts.en.yml
1538:          stage: {}
2742:          stage: {}
2747:          stage: {}
3924:          stage: {}
...
```

**Comment by Hamms** [comment]
> There's a few more instances of stage: in scripts.en.yml, are those from something else? Do they matter?

This change was to remove the plural key `stages`; I'm not sure where those keys are coming from, but I don't think they matter and they're definitely out of scope for this change.

**Review by uponthesun — APPROVED** [review]

**Comment by davidsbailey** [comment]
woohoo, glad to see these gone!

**Pull request merged**